File: day:27/minDepthTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {
    if(root === null) return 0;
    return findMinimumPath(root);
};

function findMinimumPath(node) {
    if(node === null) return null;
    let left = findMinimumPath(node.left);
    let right = findMinimumPath(node.right);
    
    if(left === null && right === null) return 1;
    
    if(left === null) return 1 + right;
    if(right === null) return 1 + left;
    
    return 1 + Math.min(left, right);
}

File: day:27/minDepthTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: TreeNode) -> int:
        if(root is None): return 0;
        return self.findMinimumPath(root);
        
    def findMinimumPath(self, node):
        if(node is None): return None;
        left = self.findMinimumPath(node.left);
        right = self.findMinimumPath(node.right);
    
        if(left is None and right is None): return 1;
    
        if(left is None): return 1 + right;
        if(right is None): return 1 + left;
    
        return 1 + min(left, right);
